src/main/java/htsjdk/tribble/AsciiFeatureCodec.java

```diff
@@ -49,14 +50,19 @@ public boolean isDone(final LineIterator lineIterator) {
     }
 
     @Override
-    public LocationAware makeIndexableSourceFromStream(final InputStream bufferedInputStream) {
-        final PositionalBufferedStream pbs;
-        if (bufferedInputStream instanceof PositionalBufferedStream) {
-            pbs = (PositionalBufferedStream) bufferedInputStream;
+    public LocationAware makeIndexableSourceFromStream(final InputStream inputStream) {
+        if (inputStream instanceof BlockCompressedInputStream) {
+            // For block compressed inputs, we need to ensure that no buffering takes place above the input stream to
+            // ensure that the correct (virtual file pointer) positions returned from this stream are preserved for
+            // the indexer. We can't used AsciiLineReader in this case since it wraps the input stream with a
+            // PositionalBufferedInputStream.
+            return new AsciiLineReaderIterator(new BlockCompressedAsciiLineReader((BlockCompressedInputStream) inputStream));
+        } else if (inputStream instanceof PositionalBufferedStream) {
+            // if this is already a PositionalBufferedStream, don't let AsciiLineReader wrap it with another one...
+            return new AsciiLineReaderIterator(new AsciiLineReader((PositionalBufferedStream)inputStream));
         } else {
-            pbs = new PositionalBufferedStream(bufferedInputStream);
+            return new AsciiLineReaderIterator(new AsciiLineReader(inputStream));
```

Comment:
Possibly; it depends on the type of the inputStream. I left this in since it represents the previous behavior for this method. If inputStream isn't wrapping a block compressed stream, and reports it's position correctly, I think it would work. We could throw instead, but if we we're going to do that, we could deprecate this method and replace it method with two overloads with the correct stream argument types.